extra/onnx.py

```diff
@@ -114,7 +116,7 @@ def run_onnx(inputs={}, debug=False):
       if inp.name in tensors: continue
       shape = type_parse(inp.type)
       # this works for gpt3 but does not work elsewhere
-      # if len(shape) >= 1 and shape[0] == 0 and shape != (0,): shape = tuple([1]+list(shape[1:]))   # 1 batch size
+      if len(shape) >= 1 and shape[0] == 0 and shape != (0,): shape = tuple([1]+list(shape[1:]))   # 1 batch size
```

Comment:
Simpler and same behaviour 
```suggestion
      if len(shape) > 1 and shape[0] == 0: shape = (1,)+shape[1:]   # 1 batch size
```

Reply:
I don't think reshaping 0 to 1 works... (I may be wrong) 
There are a lot of other type_proto that have really weird structures and those fail when reshaping like this... 

What I have now is:
optional_type has element_type = tensor_type or sequence_type
and sequence_type has element_type = tensor_type

so when tensor_type does not have dim_value, I assume it's a variable tensor and skip the shape assert:
`assert input_shape == shape, f"wrong shape for input {inp.name}, {input_shape} isn't {shape}"`
(since checking if shape is equal when type_proto doesn't provide a shape is probably not necessary)